certora/harness/ATokenWithPool.sol

```diff
@@ -0,0 +1,333 @@
+pragma solidity 0.8.10;
+pragma experimental ABIEncoderV2;
+
+import {DummyERC20Impl} from "./DummyERC20Impl.sol";
+import {IATokenWithPool} from "./IATokenWithPool.sol";
+import {SafeCast} from "@aave/core-v3/contracts/dependencies/openzeppelin/contracts/SafeCast.sol";
+import {WadRayMath} from "@aave/core-v3/contracts/protocol/libraries/math/WadRayMath.sol";
+import {IAaveIncentivesController} from "../../contracts/l1/interfaces/IAaveIncentivesController.sol";
+import {ILendingPool} from "../../contracts/l1/interfaces/ILendingPool.sol";
+
+contract ATokenWithPool is IATokenWithPool {
+    using WadRayMath for uint256;
+    using SafeCast for uint256;
+
+    /**
+    * @dev Only pool can call functions marked by this modifier.
+    **/
+    modifier onlyPool() {
+    require(msg.sender == address(POOL), "CALLER_MUST_BE_POOL");
+    _;
+    }
+
+    /**
+    * @dev UserState - additionalData is a flexible field.
+    * ATokens and VariableDebtTokens use this field store the index of the
+    * user's last supply/withdrawal/borrow/repayment. StableDebtTokens use
+    * this field to store the user's stable rate.
+    */
+    struct UserState {
+        uint128 balance;
+        uint128 additionalData;
+    }
+    // Map of users address and their state data (userAddress => userStateData)
+    mapping(address => UserState) internal _userState;
+    // Total supply
+    uint256 t;
+    // Allowances
+    mapping(address => mapping(address => uint256)) a;
+
+    string public name;
+    string public symbol;
+    uint256 public decimals;
+    address internal _treasury;
+    address internal _underlyingAsset;
+    IAaveIncentivesController public _incentivesController;
+    ILendingPool public POOL;
+
+    constructor(address pool)
+    {
+        POOL = ILendingPool(pool);
+    }
+
+    /**
+   * @notice Returns the address of the Incentives Controller contract
+   * @return The address of the Incentives Controller
+   **/
+    function getIncentivesController() external view returns (IAaveIncentivesController) {
+    return _incentivesController;
+    }
+
+    function totalSupply_super() public view virtual returns (uint256) {
```

Comment:
It's pure semantics: I've tried to mimic the original code, when super.totalSupply() was used.
Since I removed all inheritance from this code, this declaration restores the original meaning.